### DISCUSSION TITLE
Issue with folding in UI response

### DISCUSSION THREAD

**Discussion by rogeriofrsouza** [mistweaverco/kulala.nvim#268] [Q&A] [answered]
Hello 👋🏻 
I am experiencing an issue in the response UI window, seems like folding is not working. 
Was there any recent changes that affected it? It was working in the past days.

![image](https://github.com/user-attachments/assets/9023177e-dcef-44a8-9992-ac4067ea3fe4)

Setup:
NVIM v0.10.0
Build type: Release
LuaJIT 2.1.1713484068

LazyVim 12.42.0

**Comment by gorillamoe** [comment]
Maybe a regression due to https://github.com/mistweaverco/kulala.nvim/pull/249

**Reply by gorillamoe** [reply]
@wenjinnn, might know better

**Comment by wenjinnn** [comment]
I upgrade to newest commit and tested it, still works fine for me.
![image](https://github.com/user-attachments/assets/41668031-42d9-4098-8f58-d1045e3228ce)
@rogeriofrsouza could you offer more detail? Like how do you configure your `foldexpr` and do you installed json and html treesitter parser?
The correct configuration should be
```lua
vim.opt.foldmethod = 'expr'
vim.opt.foldexpr = 'v:lua.vim.treesitter.foldexpr()'
```

**Reply by rogeriofrsouza** [reply]
I have both html and json parsers installed.

The foldexpr in LazyVim depends on a [util function](https://github.com/LazyVim/LazyVim/blob/13a4a84e3485a36e64055365665a45dc82b6bf71/lua/lazyvim/util/ui.lua#L200), it's returning '0' because the UI buffer is a `nofile`

![image](https://github.com/user-attachments/assets/319a8bb0-d3d2-40b6-8904-3c5b3d3bd527)

**Reply by gorillamoe** [reply]
Strange, because it has always been a nofile right from V1 🤔 I'll check on that tomorrow

**Comment by gorillamoe** [comment, marked as answer]
Should be fixed in main due to: https://github.com/mistweaverco/kulala.nvim/pull/279

Can you check @rogeriofrsouza ?

**Reply by rogeriofrsouza** [reply]
Yes, it's fixed for me. Thank you! 🎉
